### ISSUE TITLE
`getPath` does not work with grouped routes

### ISSUE THREAD

**Issue by NuroDev** [honojs/hono#2758]
### What version of Hono are you using?

4.3.9

### What runtime/platform is your app running on?

Bun & Deno

### What steps can reproduce the bug?

I have created a basic reproducible sandbox to demonstrate the issue using Deno Deploy: https://dash.deno.com/playground/hono-grouping-getpath-bug.

## Works

```ts
import { Hono } from 'https://deno.land/x/hono@v4.3.9/mod.ts';

const groupApp = new Hono().get('/:foo/:bar', (c) => c.json({
	ok: true,
	params: c.req.param()
}));

const app = new Hono({
	// Adding the path override at the top level works
	getPath: (req) => new URL(decodeURIComponent(req.url)).pathname,
})
	.get('/', (c) => c.json({ ok: true, }))
	.route('/', groupApp);

Deno.serve(app.fetch);
```

### Does not work

```ts
import { Hono } from 'https://deno.land/x/hono@v4.3.9/mod.ts';

const groupApp = new Hono({
	// But if you try and override / modify the path in a grouped route it doesn't work
	getPath: (req) => new URL(decodeURIComponent(req.url)).pathname
})
.get('/:foo/:bar', (c) => c.json({
	ok: true,
	params: c.req.param()
}))

const app = new Hono()
  .get('/', (c) => c.json({ ok: true, }))
  .route('/', groupApp);

Deno.serve(app.fetch);
```

In summary: If you try and use `getPath` inside of a nested or grouped route, the mutation / path override does not seem to be applied.

The context here is that this is a URL that is hit by the npm CLI so all URL's are URI encoded & this `getPath` will decode them for Hono to use normally.

### What is the expected behavior?

Ideally `getPath` should apply the path modification / override at all levels, not just the top.

The alternative here would be use some kind of middleware but I think it would be better to keep with the built-ins where possible, which in this case is `getPath`.

### What do you see instead?

If you try to navigate to `/foo%2fbar` or any other URI encoded URL it will just return a 404 as the expected path does not exist.

**Comment by monoald** [comment]
Hi Nurodev, my guess is that the URL decoding has to be done at the main router because it is the first to get the path. It needs the decoded URL to know where to send the request. If not decoded, it can't match any existing route handler and then responds with a 404.

As you said, maybe a middleware would be nice to have, but it would have to be used at the main router.

**Comment by szmarczak** [comment]
FYI this is gonna be fixed by #2714

**Comment by NuroDev** [comment]
Ah okay, amazing! Thank you @szmarczak 🙏
